Zend/zend_vm_def.h

```diff
@@ -7032,6 +7032,31 @@ ZEND_VM_COLD_CONST_HANDLER(169, ZEND_COALESCE, CONST|TMP|VAR|CV, JMP_ADDR)
 	ZEND_VM_NEXT_OPCODE();
 }
 
+ZEND_VM_HOT_NOCONST_HANDLER(195, ZEND_JMP_NULL, CONST|TMPVAR|CV, JMP_ADDR)
+{
+	USE_OPLINE
+	zval *val;
+
+	val = GET_OP1_ZVAL_PTR_UNDEF(BP_VAR_R);
+
+	if (Z_TYPE_INFO_P(val) != IS_NULL) {
+		ZEND_VM_NEXT_OPCODE();
+	} else {
+		zval *result = EX_VAR(opline->result.var);
+
+		if (EXPECTED(opline->extended_value == 0)) {
+			ZVAL_NULL(result);
+		} else if (EXPECTED(opline->extended_value == (1 << 0))) {
```

Comment:
Use the `ZEND_SHORT_CIRCUITING_SCOPE_ISSET|EMPTY` constants.

Reply:
Is this just an optimization, or is there some reason it needs to be integrated in this opcode?

Reply:
(This should use constants, there's a TODO for that). Yes, this is needed for `ISSET` and `EMPTY`. It's necessary because their opcodes combine the fetching and assigning of the result so there's nowhere to jump to that would still set the result zval to the appropriate value. Hope that makes sense.